src/services/member/service.ts

```diff
@@ -94,6 +94,10 @@ export class MemberService {
     });
   }
 
+  async deleteCurrent(member: Member, { memberRepository }: Repositories) {
```

Comment:
Instead of asking the whole member, you can ask for only an Id.

It makes this function easier to call. We wouldn't need to `SELECT` the whole `Member` on the `database` before calling this function.
It would be easier to refactor (B.e : Converting all members into Account)
It reduces the possibilities of a bad implementation of this function for the future.

Reply:
I see, do you prefer:
```ts
async deleteCurrent(member: { id: string }, { memberRepository }: Repositories) {
```

or 

```ts
async deleteCurrent(memberId: string, { memberRepository }: Repositories) {
```

Reply:
I'd say 
```typescript
async deleteCurrent(memberId: string, { memberRepository }: Repositories) {
```

But I'm open to the first proposal which is interesting !

Reply:
okay ! Yes, in the end I used the second approach as I think it is less ambiguous that a string is required. The first one is "cool" because at the call site you can pass anything that contains an is prop that is a string, so it is very flexible. But the risk is that you spread the object, and then you might get un-expected consequences, mainly it might contain more than what you think.